src/dbms/session_context_handler.hpp

```diff
@@ -0,0 +1,611 @@
+    std::lock_guard<LockT> wr(lock_);
+    return sessions_.erase(session.UUID()) > 0;
+  }
+
+  /**
+   * @brief Delete database.
+   *
+   * @param db_name database name
+   * @return DeleteResult error on failure
+   */
+  DeleteResult Delete(const std::string &db_name) {
+    std::lock_guard<LockT> wr(lock_);
+    if (db_name == kDefaultDB) {
+      // MSG cannot delete the default db
+      return DeleteError::DEFAULT_DB;
+    }
+    // Check if db exists
+    try {
+      auto sc = Get_(db_name);
+      // Check if a session is using the db
+      if (!sc.interpreter_context->interpreters->empty()) {
+        return DeleteError::USING;
+      }
+    } catch (UnknownDatabase &) {
+      return DeleteError::NON_EXISTENT;
+    }
+
+    // High level handlers
+    for (auto &[_, s] : sessions_) {
+      if (!s.OnDelete(db_name)) {
+        spdlog::error("Partial failure while deleting database \"{}\".", db_name);
+        defunct_dbs_.emplace(db_name);
+        return DeleteError::FAIL;
+      }
+    }
+
+    // Low level handlers
+    const auto storage_path = StorageDir_(db_name);
+    MG_ASSERT(storage_path, "Missing storage for {}", db_name);
+    if (!interp_handler_.Delete(db_name) || !storage_handler_.Delete(db_name)) {
+      spdlog::error("Partial failure while deleting database \"{}\".", db_name);
+      defunct_dbs_.emplace(db_name);
+      return DeleteError::FAIL;
+    }
+
+    // Remove from auth
+    auth_->Lock()->DeleteDatabase(db_name);
+    // Remove from durability list
+    if (durability_) durability_->Delete(db_name);
+
+    // Delete disk storage
+    if (delete_on_drop_) {
+      std::error_code ec;
+      (void)std::filesystem::remove_all(*storage_path, ec);
+      if (ec) {
+        spdlog::error("Failed to clean disk while deleting database \"{}\".", db_name);
+        defunct_dbs_.emplace(db_name);
+        return DeleteError::DISK_FAIL;
+      }
+    }
+
+    // Delete from defunct_dbs_ (in case a second delete call was successful)
+    defunct_dbs_.erase(db_name);
+
+    return {};  // Success
+  }
+
+  /**
+   * @brief Set the default configurations.
+   *
+   * @param configs storage, interpreter and authorization configurations
+   */
+  void SetDefaultConfigs(Config configs) {
+    std::lock_guard<LockT> wr(lock_);
+    default_configs_ = configs;
+  }
+
+  /**
+   * @brief Get the default configurations.
+   *
+   * @return std::optional<Config>
+   */
+  std::optional<Config> GetDefaultConfigs() const {
+    std::shared_lock<LockT> rd(lock_);
+    return default_configs_;
+  }
+
+  /**
+   * @brief Return all active databases.
+   *
+   * @return std::vector<std::string>
+   */
+  std::vector<std::string> All() const {
+    std::shared_lock<LockT> rd(lock_);
+    return storage_handler_.All();
+  }
+
+  /**
+   * @brief Return the number of vertex across all databases.
+   *
+   * @return uint64_t
+   */
+  std::tuple<uint64_t, uint64_t, uint64_t> Info() const {
```

Comment:
Can this be a named type? `std::tuple<uint64_t, uint64_t, uint64_t>` is non descriptive.

Reply:
Sure